### ISSUE TITLE
Linux audio glitchy

### ISSUE THREAD

**Issue by mattrixed** [GrandOrgue/grandorgue#1272]
Ubuntu 22.04
GO 3.9.1

No matter the audio settings even simple stereo or multichannel audio is glitchy almost like not enough polyphony!?
RT or PA and audio buffer settings of 128 256 512 1024 doesn't fix.

**Comment by mattrixed** [comment]
may not be just GO related - tried another usb audio device simple stereo radial pro and even youtube is glitchy....
sigh

**Comment by rousseldenis** [comment]
@mattrixed what do you have as soundcard? And which hardware do you have ?

**Comment by mattrixed** [comment]
Focusrite 18i20 2nd
Radial Pro usb

This pc setup for Linux Lenovo M93P
32gb ram
Ssd
I5 4570
12743 geekbench 4 points

**Comment by rousseldenis** [comment]
Could you check syslog when playing YouTube video?

**Comment by mattrixed** [comment]
im linux noob where do I go for that?

**Comment by rousseldenis** [comment]
Open a terminal and launch this command:

`tail -f /var/log/syslog`

Then, try to reproduce the problem and look at the log in parallel.

**Comment by oleg68** [comment]
> audio is glitchy

1. Could you explain more. What symptoms of `the glitchy`?
2. Please post the screenshot of the audio settings tab (with Atl+PrintScreen)
3. Please post the screenshot of the available device screen in the audio serrings (When you are trying to change the current device)

**Comment by mattrixed** [comment]
Im at work so can't share screen right this very second but honestly same setup every OS i run linux / mac / windows  48khz 256 buffer Asio/RT/PA linux uses asla?   

With linux RT has a lot more latency up at 23 compared to 10 of PA/asla !? 256 buffer

Audio glitchy - as you are playing notes it doubles up or plays an extra squeak especially on fast or simple big chord changes - sounds like cpu underpowered or buffer settings crackling but as stated fast system and all buffer settings tried with no change....   Little glitch squeak in between notes/chords.  Plays note then glitch plays before next chord like system overun cpu busy doing other things (nothing else running) whole system for GO only...!

Tested with 2 different usb sound cards and then as stated even youtube does it so its system wide.... All audio not as pronounced as GO 
Clean install 22.04 !!  will try a different clean install on a slightly different machine... this was a M93P lenovo will try a Dell 9020

2 will do

3 will do - however something i've clicked on all that's available is the onboard in the system...  It did come up with the 18i20 but i clicked on only use as input but then its disappeared altogether from the system - GO can still see it but system can only see onboard audio or the radial pro usb (stereo out only) that i've connected also for testing...

Linux does it in stereo and multichannel no difference..... ALL sound drivers...

How is everyone getting by or it because people are on old version of GO - where as i upgrade everything all the time.... system OS and GO versions - my systems aren't so strange surely and 18i20 is a very common use product for this purpose in the land of HW/GO!?  Surely im not the only one who is having all these issues...!?  Or do people not use asio....?!  thereby not needing it as only need stereo.....!

**Comment by rousseldenis** [comment]
I  don't remember if I tried first a normal Ubuntu.

But for my current settings, I'm using Ubuntu 22 but with low-latency kernel and never had the problem you describe.

But this sounds to me like a hardware problem (as it's spread across applications)

**Comment by mattrixed** [comment]
Ye that’s what I’m using normal unbuntu 22 normal Ubuntu 22.04.1 latest….  Where you getting the distro from?  Link?

there was/is a low latency one for audio I had ver18 of that studio or something
Unless your meaning light Ubuntu? Xubintu?


All hardware though!  2 separate sound usb devices system wide…

will try another distro….

**Comment by rousseldenis** [comment]
It's a normal Ubuntu install.

The low latency kernel is installed simply through apt.

I followed this (in french) https://doc.ubuntu-fr.org/linux-lowlatency

**Comment by oleg68** [comment]
@mattrixed

There are a lot of complains mentioned by you there. We can not resolve all of them at once. I suggest to defer the latency issues for the future. Lets at the first force GO working at all and only at the second try to reduce latency.

No I don't know does the glitchy relate to the audio settings or not. It may be midi issues instead of audio.

As I understand, there are two sond devices on your system: the onboard one and 18i20, aren't they? Let's start with onboard and stereo.

Please 
1. Remove the `GrandOrgueConfig` file from your home directory. It will reset all the setting.
2. Select the Onboard Audio in the system panel. Test it with the `Test` button in the system panel.
3. Start GrandOrgue and load the Demo Organ.
4. The audio settings must be 44100/1024, `PortAudio: ALSA: default`.
5. Enable some stops and try to click on the keys with the mouse. Are they sound? Are there any glitches?
6. If no glitches, try to enable all stops (tutti) and try to play with mouse. If the volume overloading occures (red in the toolbar), reduce the volume from -15 to, for example, -30 and repeat the test
7. If everything works, then assign midi to the manuals and pedals and try to play. I understand, that it might be hard to play fast with a big latency, but we are searching the source of the problems.
8. If the glitches ocure at the previous step, try to play a midi file instead of playing with your console.

If something goes wrong, please tell me what steps succeeded and what step failed exactly.

**Comment by mattrixed** [comment]
Searching Google this a long time issue apparently and is system wide as I've found out - nothing to do with GO
Many posts with people unable to sort out or find concrete fixes or answers - sometimes changing pulse to 48 via terminal helps didn't do anything for me...

Have tried booting in wayland/normal/xorg didn't change anything...

Usb audio missing - i have that
Usb audio garbled - i have that
Usb audio crackling
Onboard works fine!  its to do with usb interface (does sound audio buffer mismatch etc)
etc....

May try another distro install but definitely not GO related or fixable on GO's behalf.  Unless you all are geniuses above what i already know you are and can fix it for the linux community!!!

**Comment by oleg68** [comment]
> May try another distro install 

Evreything in linux is fixable with any distro. So I'd not to be in a hurry.

> Usb audio missing - i have that
> Usb audio garbled - i have that
> Usb audio crackling

Usb audio works perfecly in linux.

>  changing pulse to 48

Please don't extra complex steps. Pulse is required now only for testing. Usual low latency GO setup bypasses Pulse.

If you are ready to go with USB, I can continue assisting you. First of all, please switch the sound in your system panel to usb focusrite and chek that it works (click test) and try to listen some music with any applicaton or i youtube. If everything works we can start testing with GrandOrgue.

**Comment by mattrixed** [comment]
I have moved system into office so i can dabble with this - testing quite a bit last night nothing fixed it.

I have a 2 ch Radial Pro usb audio interface.... connected to it for testing
Onboard all good
Anything usb not good 18i20 / Radial pro etc...
I've tried installing studio package lower latency etc that didn't work either...

System audio can no longer see the focusrite.... that is one of the issues online.... GO can still see it so thats fine.

Staying with the Radial Pro for testing - set to system it's all a garbled mess youtube etc... has a distinct buffer/glitchy audio issues...

Ready...

**Comment by oleg68** [comment]
> Onboard all good

Does GO work with onboard without any glitches?

> System audio can no longer see the focusrite.... that is one of the issues online.... 

1. Does the system see the Radial Pro?
2. GO may lock some audio device with some settings, so the system stops seeing it. You have to logout/logon the user session or even restart linux for being able to use it again. Please restart linux and **do not start grandorgue**. Can you select this device in the system panel?

**Comment by mattrixed** [comment]
Yes onboard with fiddling around to find best settings glitch free yes

1 yes
2 no it’s gone!  Logged out and restarted many times as per above I clicked on use as import only it was there. I then clicked on uses output again and it went. I am not worried about it for the purposes of this test. The radial is same scenario i.e. USB glitchy

**Comment by oleg68** [comment]
> The radial is same scenario i.e. USB glitchy

Does it glitchy when you are listen to some music or youtube?

**Comment by mattrixed** [comment]
Yes any sound from system or app browser YouTube GO when selecting usb audio output it’s glitchy

**Comment by mattrixed** [comment]
Fix system that will fix GO is my thought!

**Comment by oleg68** [comment]
OK, then I transfer this issue to discussion.
